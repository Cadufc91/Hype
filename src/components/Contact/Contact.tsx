import * as React from 'react';
import { 
    ContactContainer,
    InputContainer,
    FormTitle,
    Form,
    FormTop,
    FullName,
    Email,
    Subject,
    TextArea,
    SubmiBtn,
    GitContainer,
    ContactTitle,
    ContactCards,
    ContactItem,
    ContactInfo,
    SocialMedia,
    Github,
    LinkedIn,
    Facebook,
    Instagram,
} from './style';
import { IoPaperPlaneOutline } from 'react-icons/io5';
import { FiMapPin, FiMail, FiPhone,  } from 'react-icons/fi';
import { AiOutlineGithub, AiFillLinkedin } from 'react-icons/ai';
import { FaFacebookF, FaInstagram } from 'react-icons/fa';

export function Contact () {
    return(
        <ContactContainer id='contact'>
            <InputContainer>
                <FormTitle>Me mande uma mensagem aqui.</FormTitle>
                <Form>
                    <FormTop>
                        <FullName type="text" placeholder='Nome completo' />
                        <Email type="email" placeholder='Seu e-mail'/>
                    </FormTop>
                    <Subject type="text" placeholder='Assunto'/>
                    <TextArea className='message' placeholder='Mensagem'></TextArea>
                </Form>
                <SubmiBtn>Enviar Mensagem <IoPaperPlaneOutline size={20}/></SubmiBtn>
            </InputContainer>
            <GitContainer>
                <ContactTitle>Entre em contato</ContactTitle>
                <ContactCards>
                    <ContactItem>
                        <FiMapPin size={32} color="#ED1CA6"/>
                        <ContactInfo>Quissam√£, Rio de Janeiro -	28735-000, Brazil.</ContactInfo>
                    </ContactItem>
                    <ContactItem>
                        <FiMail size={32} color="#ED1CA6"/>
                        <ContactInfo>
                            dev.cadu@gmail.com
                            cadu.dev@gmail.com
                        </ContactInfo>
                    </ContactItem>
                    <ContactItem>
                        <FiPhone size={24} color="#ED1CA6"/>
                        <ContactInfo>
                            +55 955 258 2699 <br></br>
                            +55 955 100 9449
                        </ContactInfo>
                    </ContactItem>
                    <SocialMedia>
                        <Github>
                            <AiOutlineGithub size={30} />
                        </Github>
                        <LinkedIn>
                            <AiFillLinkedin size={30}/>
                        </LinkedIn>
                        <Facebook>
                            <FaFacebookF size={30}/>
                        </Facebook>
                        <Instagram>
                            <FaInstagram size={30}/>
                        </Instagram>
                    </SocialMedia>
                </ContactCards>
            </GitContainer>
        </ContactContainer>
    )
};

export default Contact;