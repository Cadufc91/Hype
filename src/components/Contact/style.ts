import styled from "styled-components";

export const ContactContainer = styled.section`
    margin-top: 100px;
    display: flex;
    flex-direction: row;
    margin-bottom: 108px;
    align-items: center;
    justify-content: center;
`;

export const InputContainer = styled.div`
    background: #fff;
    max-width: 50%;
    width: 45%;
    margin-right: 24px;
    heigth: 536px;
    border-radius: 8px;
`;

export const FormTitle = styled.h3`
    font-size: 30px;
    font-weight: 500;
    margin-left: 48px;
`;

export const Form = styled.form`
    height: 288px;
    max-width: 87%;
    display: flex;
    flex-direction: column;
    justify-content: center;
`;

export const FormTop = styled.div`
    display: flex;
    flex-direction: row;
    max-width: 100%;
    width: 100%;
    margin-left: 48px;
`;

export const FullName = styled.input`
    margin-right: 16px;
    height: 52px;
    border: 1px solid #DFE3EB;
    width: 50%;

    ::placeholder{
        font-size: 16px;
        font-weight: 400;
        color: #A5A5A5;
        padding-left: 20px;
    }
`;

export const Email = styled.input`
    height: 52px;
    border: 1px solid #DFE3EB;
    width: 50%;

    ::placeholder{
        font-size: 16px;
        font-weight: 400;
        color: #A5A5A5;
        padding-left: 20px;
    }
`;

export const Subject = styled.input`
    width: 100%;
    margin-left: 48px;
    margin-top: 16px;
    border: 1px solid #DFE3EB;
    height: 52px;

    ::placeholder{
        font-size: 16px;
        font-weight: 400;
        color: #A5A5A5;
        padding-left: 20px;
    }
`;

export const TextArea = styled.textarea`
    width: 100%;
    margin-left: 48px;
    margin-top: 16px;
    height: 120px;
    border: 1px solid #DFE3EB;

    ::placeholder{
        font-size: 16px;
        font-weight: 400;
        color: #A5A5A5;
        padding-left: 20px;
    }
`;

export const SubmiBtn = styled.button`
    margin-top: 32px;
    margin-left: 48px;
    margin-bottom: 48px;
    width: 45%;
    height: 64px;
    text-transform: uppercase;
    padding-left: 32px;
    font-size: 16px;
    color: #fff;
    border: none;
    background: #7D00FF;
    display: flex;
    align-items: center;
    gap: 5px;
    cursor: pointer;
`;

export const GitContainer = styled.div`
    width: 27%;
    height: 536px;
    background: #fff;
    border-radius: 8px;
`;

export const ContactTitle = styled.h3`
    margin-left: 48px;
    font-size: 32px;
    font-weight: 500;
`;

export const ContactCards = styled.div``;

export const ContactItem = styled.div`
    display: flex;
    flex-direction: row;
    max-width: 63%;
    margin-left: 48px;
    align-items: center;
`;

export const ContactInfo = styled.p`
    margin-left: 20px;
    font-size; 18px;
    height: 56px;
`;

export const SocialMedia = styled.div`
    max-width: 78%;
    display: flex;
    flex-direction: row;
    justify-content: center;
    gap: 12px;
    margin-left: 48px;
    margin-top: 40px;
`;

export const Github = styled.button`
    border-radius: 50%;
    background: #DFE3EB;
    border: none;
    padding: 10px;
    cursor: pointer;
    transition: all 0.3s ease-in-out;

    :hover{
        color: #fff;
        background: #7D00FF;
    }
`;

export const LinkedIn = styled.button`
    border-radius: 50%;
    background: #DFE3EB;
    border: none;
    padding: 10px;
    cursor: pointer;
    transition: all 0.3s ease-in-out;

    :hover{
        color: #fff;
        background: #7D00FF;
    }
`;

export const Facebook = styled.button`
    border-radius: 50%;
    background: #DFE3EB;
    border: none;
    padding: 10px;
    cursor: pointer;
    transition: all 0.3s ease-in-out;

    :hover{
        color: #fff;
        background: #7D00FF;
    }
`;

export const Instagram = styled.button`
    border-radius: 50%;
    background: #DFE3EB;
    border: none;
    padding: 10px;
    cursor: pointer;
    transition: all 0.3s ease-in-out;

    :hover{
        color: #fff;
        background: #7D00FF;
    }
`;