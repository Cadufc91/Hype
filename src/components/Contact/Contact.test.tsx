import { render } from "@testing-library/react";
import  { Contact } from './Contact';

describe('Header', () => {
    it('should match snapshot', () => {
        const { container } = render(<Contact />)

        expect(container.firstChild).toMatchSnapshot()
    })
})