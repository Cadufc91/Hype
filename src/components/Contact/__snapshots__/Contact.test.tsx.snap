// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<section
  class="sc-bczRLJ hQauag"
  id="contact"
>
  <div
    class="sc-gsnTZi wKRAX"
  >
    <h3
      class="sc-dkzDqf cOeTtF"
    >
      Me mande uma mensagem aqui.
    </h3>
    <form
      class="sc-hKMtZM cuUbhM"
    >
      <div
        class="sc-eCYdqJ cVMirJ"
      >
        <input
          class="sc-jSMfEi eLLcBV"
          placeholder="Nome completo"
          type="text"
        />
        <input
          class="sc-gKXOVf kYfhjL"
          placeholder="Seu e-mail"
          type="email"
        />
      </div>
      <input
        class="sc-iBkjds eJKJvC"
        placeholder="Assunto"
        type="text"
      />
      <textarea
        class="sc-ftvSup hxizHh message"
        placeholder="Mensagem"
      />
    </form>
    <button
      class="sc-papXJ hhNlf"
    >
      Enviar Mensagem 
      <svg
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M53.12 199.94l400-151.39a8 8 0 0110.33 10.33l-151.39 400a8 8 0 01-15-.34l-67.4-166.09a16 16 0 00-10.11-10.11L53.46 215a8 8 0 01-.34-15.06zM460 52L227 285"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="32"
        />
      </svg>
    </button>
  </div>
  <div
    class="sc-jqUVSM gBleNi"
  >
    <h3
      class="sc-kDDrLX keTtEA"
    >
      Entre em contato
    </h3>
    <div
      class="sc-iqcoie"
    >
      <div
        class="sc-crXcEl dvVyRJ"
      >
        <svg
          color="#ED1CA6"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(237, 28, 166);"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
          />
          <circle
            cx="12"
            cy="10"
            r="3"
          />
        </svg>
        <p
          class="sc-evZas jdFHZG"
        >
          Quissam√£, Rio de Janeiro - 28735-000, Brazil.
        </p>
      </div>
      <div
        class="sc-crXcEl dvVyRJ"
      >
        <svg
          color="#ED1CA6"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(237, 28, 166);"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
          />
          <polyline
            points="22,6 12,13 2,6"
          />
        </svg>
        <p
          class="sc-evZas jdFHZG"
        >
          dev.cadu@gmail.com cadu.dev@gmail.com
        </p>
      </div>
      <div
        class="sc-crXcEl dvVyRJ"
      >
        <svg
          color="#ED1CA6"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(237, 28, 166);"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
          />
        </svg>
        <p
          class="sc-evZas jdFHZG"
        >
          +55 955 258 2699 
          <br />
          +55 955 100 9449
        </p>
      </div>
      <div
        class="sc-breuTD gQVHmJ"
      >
        <button
          class="sc-ksZaOG gkgYch"
        >
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
            />
          </svg>
        </button>
        <button
          class="sc-hAZoDl ihTigy"
        >
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
            />
          </svg>
        </button>
        <button
          class="sc-fnykZs igbycp"
        >
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
            />
          </svg>
        </button>
        <button
          class="sc-fEOsli dRBdDk"
        >
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</section>
`;
